### ISSUE TITLE
ElementInternals does not contain form if denoted by form attribute

### ISSUE THREAD

**Issue by lennartbuit** [calebdwilliams/element-internals-polyfill#89]
Buttons can associate to forms that they are a descendant of, or forms that have an id specified by the [`form`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-form) attribute.

So can form associated custom buttons in Chrome (107), see for example this sandbox: https://codesandbox.io/s/loving-lena-p56owk?file=/index.html . Both the button inside the form, as well as the one specified by the `form` attribute can find the form it is associated to.

However, in browsers that require this polyfill, such as Safari, only the form associated button that is a descendant of the form can properly find its form.

**Comment by calebdwilliams** [comment]
Good callout. Looks like this works generally for all APIs as well. I'll have to poke around at the best way to solve this unless you'd like to take a shot at it.

**Comment by lennartbuit** [comment]
I added a workaround to our internal DS, if `this._internals.form` is unavailable, try `document.getElementById(this.form)`. But I know too little of the specs to know whether that is _really_ correct.

**Comment by calebdwilliams** [comment]
Yeah, my concern is basically wanting to confirm how this works if the form attribute changes midstream and how any values or behavior pivot.

**Comment by calebdwilliams** [comment]
Yeah, here's an [example of what I was worried about](https://codepen.io/calebdwilliams/pen/KKeXYeP?editors=1010). Although it might be as simple as forwarding the test attribute to the hidden input … but there are likely other side effects to worry about like event listeners, etc. 

I'll play with it, but it won't be a fast fix.

**Comment by lennartbuit** [comment]
Thats alright; I think my naive workaround is enough for the foreseeable future. Let me know if theres anything I can do for you ^^.

**Comment by joelabrahamsson** [comment]
I just ran into this issue. I have a form which is submitted by a button which doesn't reside inside the form.
Maybe I'm naive but wouldn't a simple solution be to change the event listener in `wireSubmitLogic` to listen for the forms `submit` event instead of for clicks on buttons?

**Comment by calebdwilliams** [comment]
The problem is more how to tell when the form changes. It would be feasible for a developer to do something along the following lines: 

```typescript
html`
  <form action="one" id="one">
    <button type="submit">Continue</button>
  </form>

  <form action="two" id="two">
    <button type="submit">Continue</button>
  </form>

  <x-control form=${someVariable ? 'one' | 'two'}></x-control>
`;
```

When the value of `someVariable` changes we'd need to reinitialize all of the work. You might be right about the `wireSubmitLogic` code, I'll have to go back and see why/when that was changed because I remember it being a gap in my original thinking on this polyfill. 

I am taking some time off from writing code over the holiday season as I am recharging before starting my next role in January but I might get bored and try to take this on later. If someone else would like to claim it, let me know.

**Comment by enjoythelive1** [comment]
I was reading the spec for [form associated elements](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#association-of-controls-and-forms) and it seems to not be that complicated. 

Basically, when [the from attribute changes](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#association-of-controls-and-forms:~:text=When%20a%20listed%20form%2Dassociated%20element%27s%20form%20attribute%20is%20set%2C%20changed%2C%20or%20removed%2C%20then%20the%20user%20agent%20must%20reset%20the%20form%20owner%20of%20that%20element.) or [the id of any element in the tree changes](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#association-of-controls-and-forms:~:text=When%20a%20listed%20form%2Dassociated%20element%20has%20a%20form%20attribute%20and%20the%20ID%20of%20any%20of%20the%20elements%20in%20the%20tree%20changes%2C%20then%20the%20user%20agent%20must%20reset%20the%20form%20owner%20of%20that%20form%2Dassociated%20element.) (I imagine document or shadow dom where the element is), then we have to [recompute the form the element belongs too](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#reset-the-form-owner). 

So, we could follow the spec. It would require extending what the mutation observer is observing, and it is not that simple, but it is doable. 

If you guys agree, I could try making a PR to solve this issue, following the approach listed in the spec.
